pkg/scheduler/framework/plugins/default_registry.go

```diff
@@ -31,3 +31,22 @@ func NewDefaultRegistry() framework.Registry {
 		noop.Name: noop.New,
 	}
 }
+
+// Uncomment this function once the first predicate/priority config producer is registered.
+// func appendToPluginSet(pluginSet *config.PluginSet, name string, weight int32) *config.PluginSet {
+// 	if pluginSet == nil {
+// 		pluginSet = &config.PluginSet{}
+// 	}
+// 	pluginSet.Enabled = append(pluginSet.Enabled, config.Plugin{Name: name, Weight: weight})
+// 	return pluginSet
+// }
+
+func init() {
+	// Register config producers for default plugins that implement existing predicates and
+	// priorities. For example:
+	//
+	// factory.RegisterPredicateToFrameworkConfigProducer(predicates.PodFitsHostPortsPred,
+	// 	func() (plugins config.Plugins, pluginConfig []config.PluginConfig) {
```

Comment:
will we know pluginConfig at `init` time? I feel like some configurations might come from runtime sources.
In that case, you might need pluginConfig to be a parameter as well?

Reply:
Is it just the weight?
But we could have other things for #80639

Reply:
I was mainly thinking of predicates/priorities which do not require customization, all the ones supported by DefaultProvider fall in that category. I suspect that in most (may be all) cases an empty pluginConfig will be returned, unless one wants to provide a fixed configuration. For custom predicates/priorities, we will need to explicitly register their producer, and that producer function can capture whatever parameters it wants.

In any case, we can change this from ```init``` to a function that the configurator calls explicitly with whatever arguments the custom predicates/priorities needs for their config producers.

Reply:
I will remove this for now, I think it will likely need to be in the factory package.